tests/math/test_functions.py

```diff
@@ -2266,3 +2264,180 @@ def test_sort(self, input, test_output):
 
 
 ones_functions = [onp.ones, np.ones, jnp.ones, torch.ones, tf.ones]
+
+state_00 = [1, 0, 0, 0]
+state_01 = [0, 1, 0, 0]
+state_10 = [0, 0, 1, 0]
+state_11 = [0, 0, 0, 1]
+
+state_00_10 = [1, 0, 1, 0] / onp.sqrt(2)
+state_01_11 = [0, 1, 0, 1] / onp.sqrt(2)
+
+mat_00 = onp.zeros((4, 4))
+mat_00[0, 0] = 1
+
+mat_01 = onp.zeros((4, 4))
+mat_01[1, 1] = 1
+
+mat_10 = onp.zeros((4, 4))
+mat_10[2, 2] = 1
+
+mat_11 = onp.zeros((4, 4))
+mat_11[3, 3] = 1
+
+mat_0 = onp.zeros((2, 2))
+mat_0[0, 0] = 1
+
+mat_1 = onp.zeros((2, 2))
+mat_1[1, 1] = 1
+
+mat_00_10 = onp.zeros((4, 4))
+mat_00_10[0, 0] = 0.5
+mat_00_10[2, 2] = 0.5
+mat_00_10[0, 2] = 0.5
+mat_00_10[2, 0] = 0.5
+
+mat_01_11 = onp.zeros((4, 4))
+mat_01_11[1, 1] = 0.5
+mat_01_11[3, 3] = 0.5
+mat_01_11[1, 3] = 0.5
+mat_01_11[3, 1] = 0.5
+
+mat_0_1 = [[0.5, 0.5], [0.5, 0.5]]
+
+# fmt: off
+state_vectors = [
+    (state_00, (mat_0, mat_0, mat_00)),
+    (state_01, (mat_0, mat_1, mat_01)),
+    (state_10, (mat_1, mat_0, mat_10)),
+    (state_11, (mat_1, mat_1, mat_11)),
+    (onp.array(state_00), (mat_0, mat_0, mat_00)),
+    (onp.array(state_01), (mat_0, mat_1, mat_01)),
+    (onp.array(state_10), (mat_1, mat_0, mat_10)),
+    (onp.array(state_11), (mat_1, mat_1, mat_11)),
+    (np.array(state_00), (mat_0, mat_0, mat_00)),
+    (np.array(state_01), (mat_0, mat_1, mat_01)),
+    (np.array(state_10), (mat_1, mat_0, mat_10)),
+    (np.array(state_11), (mat_1, mat_1, mat_11)),
+    (jnp.array(state_00), (mat_0, mat_0, mat_00)),
+    (jnp.array(state_01), (mat_0, mat_1, mat_01)),
+    (jnp.array(state_10), (mat_1, mat_0, mat_10)),
+    (jnp.array(state_11), (mat_1, mat_1, mat_11)),
+    (torch.tensor(state_00), (mat_0, mat_0, mat_00)),
+    (torch.tensor(state_01), (mat_0, mat_1, mat_01)),
+    (torch.tensor(state_10), (mat_1, mat_0, mat_10)),
+    (torch.tensor(state_11), (mat_1, mat_1, mat_11)),
+    (tf.Variable(state_00), (mat_0, mat_0, mat_00)),
+    (tf.Variable(state_01), (mat_0, mat_1, mat_01)),
+    (tf.Variable(state_10), (mat_1, mat_0, mat_10)),
+    (tf.Variable(state_11), (mat_1, mat_1, mat_11)),
+    (tf.constant(state_00), (mat_0, mat_0, mat_00)),
+    (tf.constant(state_01), (mat_0, mat_1, mat_01)),
+    (tf.constant(state_10), (mat_1, mat_0, mat_10)),
+    (tf.constant(state_11), (mat_1, mat_1, mat_11)),
+    (state_00_10, (mat_0_1, mat_0, mat_00_10)),
+    (state_01_11, (mat_0_1, mat_1, mat_01_11)),
+]
+
+single_wires_list = [
+    [0],
+    [1],
+]
+
+multiple_wires_list = [
+    [0, 1]
+]
+# fmt: on
+
+
+class TestDensityMatrixFromStateVectors:
+    """Tests for creating a density matrix from state vectors."""
+
+    @pytest.mark.parametrize("state_vector, expected_density_matrix", state_vectors)
+    @pytest.mark.parametrize("wires", single_wires_list)
+    def test_density_matrix_from_state_vector_single_wires(

```

Comment:
As far as I can tell, we have two tests with the exact same name and docstring, but test slightly different things. I'm not sure why pytest is allowing this, but the distinction between the two different tests is a little hard to see. 


I otherwise wouldn't be a fan of this, but it might be worth putting both of these checks in the same test side by side.

```
        density_matrix = fn.quantum._density_matrix_from_state_vector(state_vector, wires=wires)
         assert np.allclose(density_matrix, expected_density_matrix[wires[0]])

        density_matrix = fn.to_density_matrix(state_vector, wires=wires)
         assert np.allclose(density_matrix, expected_density_matrix[wires[0]])
```

Reply:
Thanks I changed the names, it was a mistake!